src/wp-includes/option.php

```diff
@@ -253,6 +253,82 @@ function get_option( $option, $default_value = false ) {
 	return apply_filters( "option_{$option}", maybe_unserialize( $value ), $option );
 }
 
+/**
+ * Primes specific options into the cache with a single database query.
+ *
+ * Only options that do not already exist in cache will be primed.
+ *
+ * @since 6.4.0
+ *
+ * @global wpdb $wpdb WordPress database abstraction object.
+ *
+ * @param array $options An array of option names to be primed.
+ */
+function prime_options( $options ) {
+	$alloptions     = wp_load_alloptions();
+	$cached_options = wp_cache_get_multiple( $options, 'options' );
+
+	// Filter options that are not in the cache.
+	$options_to_prime = array();
+	foreach ( $options as $option ) {
+		if ( ( ! isset( $cached_options[ $option ] ) || ! $cached_options[ $option ] ) && ! isset( $alloptions[ $option ] ) ) {
+			$options_to_prime[] = $option;
+		}
+	}
+
+	if ( ! empty( $options_to_prime ) ) {
+		global $wpdb;
+		$option_names = implode( "','", array_map( 'esc_sql', $options_to_prime ) );
+		$query   = $wpdb->prepare( "SELECT option_name, option_value FROM $wpdb->options WHERE option_name IN (%s)", $option_names );
```

Comment:
Need to double check that the way we're handling this string replacement is correct. See: https://github.com/WordPress/WordPress-Coding-Standards/blob/dc2f21771cb2b5336a7e6bb6616abcdfa691d7de/WordPress/Tests/DB/PreparedSQLPlaceholdersUnitTest.inc#L70-L124 (related: https://core.trac.wordpress.org/ticket/54042)

Reply:
Indeed, the current approach will produce `'option1\',\'option2\',\',option3'`.

In addition, I don't believe there's any need to run `esc_sql()` on `$options_to_prime`. [$wpdb->prepare()](https://github.com/WordPress/wordpress-develop/blob/0c96ba106c5ffbf6098e172291c5bf533ae4bc07/src/wp-includes/class-wpdb.php#L1494) already performs escaping.

Given this and the coding standards example Joe has linked above, I'd suggest something like this instead:
```php
$results = $wpdb->get_results(
	$wpdb->prepare(
		sprintf(
			"SELECT option_name, option_value FROM $wpdb->options WHERE option_name IN (%s)",
			implode( ',', array_fill( 0, count( $options_to_prime ), '%s' ) )
		),
		$options_to_prime
	)
);
```

Reply:
Thank you, @joemcgill, for pointing that out. I missed that escaping, which is why my local unit tests are failing.

@costdev, thank you for the suggestion; it makes sense. I have updated it at https://github.com/WordPress/wordpress-develop/pull/4970/commits/b191e7b8d719a92c0e1f192c3360333ce5edc340.

Reply:
This looks right to me. Thanks for addressing!